sdcm/cluster.py

```diff
@@ -2788,15 +2788,28 @@ def kill_gemini_thread(self):
                 if kill_result.exit_status != 0:
                     self.log.error('Terminate gemini on node %s:\n%s', loader, kill_result)
 
-    def get_non_system_ks_cf_list(self, loader_node, db_node_ip):
+    def wrapped_get_non_system_ks_cf_list(self, db_node, request_timeout=300):
+        loader_node = None
+        while not loader_node:
+            loader_node = random.choice(self.nodes)
+            if loader_node.termination_event.isSet():
```

Comment:
Why do we need such complicated while loop here? Why do you check if termination event on the loader is set?

Reply:
I used same way how it's performed in run_fullscan_thread. If it correct there, I think it's right in other places that does same

Reply:
In fullscan it used for another purpose, since fullscan is a special thread. When `termination_event` is set that means the the test is over, but for some reason in your code you still want to select loader node, this is wrong. You don't need to add this logic here (this should be handles by Nemesis object that is currently has very simple implementation for handling `end of the test` event)

Reply:
fixed